src/Serializer/AbstractItemNormalizer.php

```diff
@@ -311,7 +311,13 @@ protected function denormalizeRelation(string $attributeName, PropertyMetadata $
             // repeat the code so that IRIs keep working with the json format
             if (true === $this->allowPlainIdentifiers && $this->itemDataProvider) {
                 try {
-                    return $this->itemDataProvider->getItem($className, $value, null, $context + ['fetch_data' => true]);
+                    $item = $this->itemDataProvider->getItem($className, $value, null, $context + ['fetch_data' => true]);
```

Comment:
It's the `itemDataProvider` that should be throwing th ` ItemNotFoundException`  not the Normalizer.

Reply:
I think about it, but it can brake backward compatibility. I actually don't know very well api-platform and where ItemDataProvider used and can't predict what happens with all such places if ItemDataProvider start throwing this exception. Is there everything will be fine?

If ok, I will make change tomorrow.

Reply:
Oh my bad you're right, seems we're using doing this in the [ReadListener](https://github.com/api-platform/core/blob/45c537fac12b579aea321ac2170f90abcfcba9ff/src/EventListener/ReadListener.php#L110).

Reply:
If it is only one place, than we can change it too. Problem only with custom data providers and who use it with "allow_plain_identifiers". They need to change their DataProviders and add throw for this exception. But since nobody complain about this, looks like it will be not a problem. But anyway, here is potential brake of BC for such developers, decision up to you, I don't know how strict you are with BC.

Reply:
ping @soyuka , so, throw exception at ItemDataProvider and change ReadListener also or leave it as is?

Reply:
Leave it as is for now.

@dunglas I thought that we could add a method in the IriConverter that would do this portion of code:

https://github.com/api-platform/core/blob/45c537fac12b579aea321ac2170f90abcfcba9ff/src/EventListener/ReadListener.php#L110-L124

We could then deduplicate some code, wdyt?